scipy/optimize/tests/test_differentiate.py

```diff
@@ -394,3 +394,119 @@ def test_saddle_gh18811(self, case):
         res = differentiate(*case, step_direction=[-1, 0, 1], atol=atol)
         assert np.all(res.success)
         assert_allclose(res.df, 0, atol=atol)
+
+
+class TestJacobian:
+
+    # Example functions and Jacobians from Wikipedia:
+    # https://en.wikipedia.org/wiki/Jacobian_matrix_and_determinant#Examples
+
+    def f1(z):
+        x, y = z
+        return [x ** 2 * y, 5 * x + np.sin(y)]
+
+    def df1(z):
+        x, y = z
+        return [[2 * x * y, x ** 2], [np.full_like(x, 5), np.cos(y)]]
+
+    f1.mn = 2, 2
+    f1.ref = df1
```

Comment:
```suggestion
    f1.mn = 2, 2  # type: ignore
    f1.ref = df1  # type:ignore
```

I looked up that the `mypy` issue is because it doesn't like adding attributes to functions. I think you can just add [type: ignore](https://mypy.readthedocs.io/en/stable/common_issues.html#spurious-errors-and-locally-silencing-the-checker) comments, since it's not really a problem to offend `mypy` in this way for short lived functions inside of tests.

Reply:
Yup, that's what I planned on doing.